### DISCUSSION TITLE
Cross-Compiling Qt 5.15.3 on Raspberry Pi 3B+ with newest RaspbianOS (Bullseye) rsults in errors.

### DISCUSSION THREAD

**Discussion by amueller11** [abhiTronix/raspberry-pi-cross-compilers#101] [Q&A]
Hi there,
many thanx for the cross-compiler and the description to use on the Raspberry Pi.

I'm still learning to program some stuff on the RPi, so maybe my questions are very stupid and easy to answer but i didn't find an answer yet.

I'm using Debian Bullseye and a RPi 3B+ with the lastest RaspbianOS (Bullseye).
After several tries i now able to cross-compile QT without any errors but using some example-programs run into errors.

I discovered 3 problems with your description so i maybe this are the reasons for my problems.

1. With the new RaspbianOS there is no choice to enable the FakeKMS as you stated out in step 4 on RPi.
2. On my newly setup RPi there ist no directory /opt/vc as used in step 6 on the Host-Machine.
3. Configuring Qt 5.15.3 stated out "EGLFS Raspberry Pi ................... yes" and i'm not sure if this is correct for the RPi 3B+ or how to change this.

The compiling Qt stated in the following error "qeglfsbrcmintegration.cpp:74:5: error: ‘EGL_DISPMANX_WINDOW_T’ was not declared in this scope" which i could fix with some Google-Research so that all compiling resulted in no errors, but running the eymple-programs on the RPi resulted in errors.

Did i do anything wrong or this there a clue to use my environment with Qt ?
Any help is welcome.

Regards
Achim

**Comment by abhiTronix** [comment]
Hi @amueller11, This issue has been already reported in issue #99. I thinking Raspberry Pi OS dropped support for FakeKMS. Therefore, we need to move away from it to compile QT properly.

**Reply by stevenm3656** [reply]
Hello, thanks for all your work on this repo. Any idea on how to do this? Currently running into a wall.

**Reply by abhiTronix** [reply]
@stevenm3656 Yes we need to use graphics drivers for EGLFS instead of FAKE KMS, but currently I don't have Raspberry Pi accessible to test this solution.
